docs/contracts/abstract-contracts.rst

```diff
@@ -6,12 +6,15 @@
 Abstract Contracts
 ******************
 
-Contracts need to be marked as abstract when at least one of their functions is not implemented.
-Contracts may be marked as abstract even though all functions are implemented.
-
-This can be done by using the ``abstract`` keyword as shown in the following example. Note that this contract needs to be
-defined as abstract, because the function ``utterance()`` was defined, but no implementation was
-provided (no implementation body ``{ }`` was given).
+Contracts must be marked as abstract when at least one of their functions is not implemented or when
+their constructors (or declarations) do not provide arguments for all of their base contract constructors.
```

Comment:
By "or declarations" I assume you mean the inheritance list? I don't think this is the best way to word it - the first thing that comes to mind for me were the state variable declarations. I'd omit that detail. Or you could link to https://docs.soliditylang.org/en/latest/contracts.html#arguments-for-base-constructors if you think this requires that detail.
```suggestion
Contracts must be marked as abstract when at least one of their functions is not implemented or when
they do not provide arguments for all of their base contract constructors.
```

Reply:
In general the docs don't do a great job of using strictly defined words for things like "declaration" vs "definition" and what I meant here is the declaration of the contract itself (as in ``contract foo is bar ...`` vs ``contract foo is bar(1)...``

Reply:
That's all the more reason not to use the somewhat ambiguous "declaration" here. If you really want to have it here I'd suggest "inheritance list". But I still think that just not going into that detail here is the best option. We have a whole separate section on the topic.